classes/class-log.php

```diff
@@ -194,7 +194,12 @@ public function is_record_excluded( $connector, $context, $action, $user = null,
 				'role'       => ( ! empty( $exclude_rule['author_or_role'] ) && ! is_numeric( $exclude_rule['author_or_role'] ) ) ? $exclude_rule['author_or_role'] : null,
 			);
 
-			$exclude_rules = array_filter( $exclude, 'strlen' );
+			$exclude_rules = array_filter(
+				$exclude,
+				static function ( $value ) {
+					return null === $value || strlen( (string) $value ) > 0;
```

Comment:
This logic now allows `null` values to be included in the filtered output:

```php
$list = [ 1, 'two', false, null, 0 ];

$rules = array_filter(
    $list,
    function ( $value ) {
        return null === $value || strlen( (string) $value ) > 0;
    }
);

var_dump( $rules );
```

returns:

```
array(4) {
  [0]=>
  int(1)
  [1]=>
  string(3) "two"
  [3]=>
  NULL
  [4]=>
  int(0)
}
```

while we would expect the `null` value to be removed.

Could we change this to `strlen( (string) $value ) > 0`?

Reply:
I think you're reading the previous logic wrong. Previously null was also allowed which is why I added the specific check for null. Removing that would actually be a change in the logic.

Reply:
@ocean90 I just checked this again -- previously it was using `array_filter( $exclude, 'strlen' )` which does _remove_ `null` values:

```php
$list = [ 1, 'two', false, null, 0 ];
var_dump( 'before', array_filter( $list, 'strlen' ) );
```

which returns:

```
Deprecated: strlen(): Passing null to parameter #1 ($string) of type string is deprecated in /Users/kaspars/Desktop/filter.php on line 12
string(6) "before"
array(3) {
  [0]=>
  int(1)
  [1]=>
  string(3) "two"
  [4]=>
  int(0)
}
```

while the updated version keeps the `null` in the list as mentioned above.